packages/core-components/src/lib/unsorted/ui/ErrorOverlay.svelte

```diff
@@ -0,0 +1,77 @@
+<script>
+	import { fade, fly } from 'svelte/transition';
+	import {
+		handleClosingTagErrors,
+		handleUnexpectedBlockClosingTagErrors,
+		handleOpenTagErrors,
+		handleExpectedCharacterErrors,
+		handlePropErrors,
+		handleUnexpectedEnd,
+		handleUnexpectedTokens,
+		handleExpectedWhitespace
+	} from '@evidence-dev/component-utilities/hmrErrorHandling';
+	let hasError = false;
+	let message;
+
+	if (import.meta.hot) {
+		import.meta.hot.on('vite:error', (data) => {
+			const err = data.err;
+
+			const errorHandlers = [
+				handleClosingTagErrors,
+				handleUnexpectedBlockClosingTagErrors,
+				handleOpenTagErrors,
+				handleExpectedCharacterErrors,
+				handlePropErrors,
+				handleUnexpectedEnd,
+				handleUnexpectedTokens,
+				handleExpectedWhitespace
+			];
+
+			let customErrorMessage = errorHandlers.reduce((acc, handler) => {
+				return acc || handler(err);
+			}, null);
```

Comment:
Are these messages always going to be in order?
e.g. could `handleExpectedWhitespace` override an error from `handleOpenTagErrors`?

If yes, we could consider making this an array and showing all of the found errors / potential solutions

This can also be made `const`

Reply:
There could be multiple of these things going wrong at once, but I'm not sure if vite would supply all of the reasons or only one at a time. I'm assuming one at a time since that's what appears in the current HMR error overlay